langchain/document_loaders/directory.py

```diff
@@ -75,7 +75,34 @@ def load(self) -> List[Document]:
         """Load documents."""
         p = Path(self.path)
         docs: List[Document] = []
-        items = list(p.rglob(self.glob) if self.recursive else p.glob(self.glob))
+
+        # Glob specified file pattern(s)
+        if self.file_pattern:
+            # Add "." prefix in case user did not include it
```

Comment:
Perhaps a good to have safety check? Because you never know if users types `pdf` instead of `.pdf`.